common/pkg/libartifact/store/reference.go

```diff
@@ -0,0 +1,61 @@
+package store
+
+import "go.podman.io/image/v5/docker/reference"
+
+// ArtifactReference is a fully qualified oci reference except for tag, where we add
+// "latest" as the tag if tag is empty.  Valid references:
+//
+// quay.io/podman/machine-os:latest
+// quay.io/podman/machine-os
+// quay.io/podman/machine-os@sha256:916ede4b2b9012f91f63100f8ba82d07ed81bf8a55d23c1503285a22a9759a1e
```

Comment:
Please document whether repo:tag@digest references are valid (and if not, make sure they are rejected)